traitsui/testing/tester/target_registry.py

```diff
@@ -288,7 +292,7 @@ def get_location_doc(self, target_class, locator_class):
             If the given locator and target types are not supported.
         """
         self._location_registry.get_value(
-            target_class=target_class,
+            target_class=target.__class__,
             key=locator_class,
         )
         # This maybe configurable in the future via register_location
```

Comment:
😄 
These changes all make a lot of sense to me.  The only methods we expect users to call are really `register_interaction` and `register_location`, for which the api has remained unchanged.  
Everything else (ie. all the `get_*` methods), I agree, are likely to not have been used externally, and really shouldn't be.

It may be slightly confusing (or surprising I guess?) that these other `TargetRegistry` methods now typically take a `target` itself rather than a `target_class` as an argument, but `register_interaction` and `register_location` still take a `target_class`, however I think that is perfectly okay.  It makes sense that registering an interaction or location is a general act, that applies to all instances of a particular class.